src/libs/merge/kdbmerge.c

```diff
@@ -0,0 +1,527 @@
+
+#include "kdbmerge.h"
+#include "kdb.h"
+#include "kdbassert.h"
+#include "kdblogger.h"
+#include "kdbprivate.h"
+#include <ctype.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+int nonOverlapOnlyBaseCounter = 0;  // Conflict where only base is different (it exists) and our and their are empty.
+int nonOverlapBaseEmptyCounter = 0; // Conflict where only base is different (it is empty) and our=their (not empty).
+int nonOverlapAllExistCounter = 0;  // All three values exist, only base is different
+int overlap3different = 0;	  // Only overlap conflicts where all three keys exist and have different values
+// always access with getter
+int overlap1empty = 0; // counts overlaps where one set is empty. Is the double of the real amount in the end.
+
+int getNonOverlapAllExistConflicts (void)
+{
+	if (nonOverlapAllExistCounter % 3 != 0)
+	{
+		fprintf (stderr, "This should be a multiple of 3 at the end of each checkSingleSet\n");
+		return -1;
+	}
+	return nonOverlapAllExistCounter / 3;
+}
+
+int getBaseEmptyConflicts (void)
+{
+	if (nonOverlapBaseEmptyCounter % 2 != 0)
+	{
+		fprintf (stderr, "This should be a multiple of 2 at the end of each checkSingleSet\n");
+		return -1;
+	}
+	return nonOverlapBaseEmptyCounter / 2;
+}
+
+int getOverlap3different (void)
+{
+	if (overlap3different % 3 != 0)
+	{
+		fprintf (stderr, "This should be a multiple of 3 at the end of each checkSingleSet\n");
+		return -1;
+	}
+	return overlap3different / 3;
+}
+
+int getOverlap1empty (void)
+{
+	if (overlap1empty % 2 != 0)
+	{
+		fprintf (stderr, "This should be a multiple of 2 at the end of each checkSingleSet\n");
+		return -1;
+	}
+	return overlap1empty / 2;
+}
+
+int getTotalOverlaps (void)
+{
+	return getOverlap1empty () + getOverlap3different ();
+}
+
+int getTotalNonOverlaps (void)
+{
+	return getBaseEmptyConflicts () + getNonOverlapAllExistConflicts () + nonOverlapOnlyBaseCounter;
+}
+
+int getTotalConflicts (void)
+{
+	return getTotalNonOverlaps () + getTotalOverlaps ();
+}
+
+void printKs (KeySet * ks)
+{
+	Key * cur = 0;
+	fprintf (stdout, "DEBUG: Iterate over all keys:\n");
+	ksRewind (ks);
+	while ((cur = ksNext (ks)) != 0)
+	{ /* Iterates over all keys and prints their name */
+		fprintf (stdout, "DEBUG: --%s\n", keyName (cur));
+	}
+}
+
+char * strremove (char * string, const char * sub)
+{
+	size_t length = strlen (sub);
+	if (length > 0)
+	{
+		char * p = string;
+		while ((p = strstr (p, sub)) != NULL)
+		{
+			memmove (p, p + length, strlen (p + length) + 1);
```

Comment:
Did you see Elektra's internal functions. There is a elektraMemmove (and also elektraStrCmp)

Reply:
But elektraMemmove takes two Key** as parameters?

Reply:
If it does not meet your requirements, you can of course not use it.